### PR TITLE
modify fhzero logic in emptying buckets

### PR DESCRIPTION
**Pull request by pjpegion**
## Description

The current logic in CCPP_driver does not allow for buckets to be emptied every time-step.
This one-line modification fixes that.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes noaa-emc/fv3atm/issues/#530

## Testing

How were these changes tested? 
running 24-hour forecasts, saving every time-step. Time-averaged fields no longer accumulate with time.

What compilers / HPCs was it tested with?  
Hera/intel
Are the changes covered by regression tests? 
No, and I am not sure if we want an RT with this.

Have the ufs-weather-model regression test been run? On what platform?  
Limited set have been run on hera.
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
-NO
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

None

### CODE REVIEW

File: ccpp/driver/CCPP_driver.F90 [hunk 1]
```diff
@@ -149,7 +149,7 @@ subroutine CCPP_step (step, nblks, ierr)
       endif
 
       !--- determine if physics diagnostics buckets need to be cleared
-      if (mod(GFS_control%kdt,GFS_control%nszero) == 1) then
+      if ((mod(GFS_control%kdt-1,GFS_control%nszero)+1) == 1) then
```
Comment (climbfuji) [inline ccpp/driver/CCPP_driver.F90]: How is that different from
```
if (mod(GFS_control%kdt,GFS_control%nszero) == 0) then
```
?

**Comment by pjpegion** [comment]
I assume going from mod 1 to mod 0 would change the baseline, since the buckets would be emptied 1 timestep earlier

**Comment by climbfuji** [comment]
> I assume going from mod 1 to mod 0 would change the baseline, since the buckets would be emptied 1 timestep earlier
> […](#)

Yes, of course, you are right.

**Review by climbfuji — APPROVED** [review]

File: ccpp/driver/CCPP_driver.F90 (hunk 1 above)
Comment (DusanJovic-NOAA) [reply ccpp/driver/CCPP_driver.F90]: How about:

`if (mod(GFS_control%kdt-1,GFS_control%nszero) == 0) then`

Comment (pjpegion) [reply ccpp/driver/CCPP_driver.F90]: @DusanJovic-NOAA that makes sense.  I will test to confirm it gives the answer I expect.

Comment (pjpegion) [reply ccpp/driver/CCPP_driver.F90]: @DusanJovic-NOAA's recommendation works.  Thanks!

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**